src/rust/lib/system/web/src/lib.rs

```diff
@@ -481,3 +483,79 @@ pub mod traits {
     pub use super::NodeInserter;
     pub use super::NodeRemover;
 }
+
+/// Sleeps for the specified amount of time.
+///
+/// This function might sleep for slightly longer than the specified duration but never less.
+///
+/// This function is an async version of std::thread::sleep.
+#[cfg(target_arch = "wasm32")]
+pub async fn sleep(duration:Duration) {
+    use wasm_bindgen_futures::JsFuture;
+
+    let performance = performance();
+    let call_time   = performance.now();
+    let future : JsFuture = js_sys::Promise::new(&mut |resolve:Function,_| {
+        let time_elapsed_from_call = ((performance.now() - call_time) * 1000.0) as i32;
+        let duration               = duration.as_millis() as i32;
+        let duration               = (duration - time_elapsed_from_call).max(0);
+        let window                 = window();
+        let err                    = "setTimeout failed";
+        window.set_timeout_with_callback_and_timeout_and_arguments_0(&resolve,duration).expect(err);
+    }).into();
+    future.await.expect("setTimeout future failed");
```

Comment:
Errors should be real sentences! :)

Reply:
@wdanilo I know these messages looks generic, but web-sys' has error for almost every single call. The reason `setTimeout` and its future fails isn't clear to me. Do you have any better suggestion for it?

Reply:
Wojtek meant that sentence should start with capital letter and end with dot.